### PR TITLE
Document HEOS group joining/unjoining

### PR DESCRIPTION
**Pull request by klada**
## Proposed change

Documents the joining and unjoining of HEOS groups (new feature)

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase:
https://github.com/home-assistant/core/pull/32568
- This PR fixes or closes issue:

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_integrations/heos.markdown [hunk 1]
```diff
@@ -63,6 +64,38 @@ Use the sign-in service to sign the connected controller into a HEOS account so
 
 Use the sign-out service to sign the connected controller out of a HEOS account. An error message is logged if sign-out is unsuccessful. There are no parameters to this service.
 
+### Service `heos.join_group`
+
+This service can be used for joining or creating a HEOS group. A HEOS group always contains exactly one leader player and at least another player which acts as a group member. All member players will synchronously play the same audio as the group leader. The group can be controlled by any player of the group. For creating a new HEOS group or joining more members to an existing group the service call looks as follows:
+
+```json
```
Comment (frenck) [inline source/_integrations/heos.markdown]: Service calls should be formated in YAML.

File: source/_integrations/heos.markdown [hunk 2]
```diff
@@ -63,6 +64,38 @@ Use the sign-in service to sign the connected controller into a HEOS account so
...
+  "leader": "media_player.heos_player1",
+  "members": [
+    "media_player.heos_player2"
+  ]
+}
+```
+
+| Attribute              | Description
+| ---------------------- | ---------------------------------------------------------------- |
+| `leader`               | The HEOS group leader. [Required]
```
Comment (frenck) [inline source/_integrations/heos.markdown]: Add a column to indicate if an attribute is required or not. Please see other parts of our documentation for an example.

File: source/_integrations/heos.markdown [hunk 3]
```diff
@@ -63,6 +64,38 @@ Use the sign-in service to sign the connected controller into a HEOS account so
...
+| `members`              | Additional media players to be included in the group. [Required]
+
+### Service `heos.unjoin_group`
+
+Use the unjoin-group service to remove a single media player from an existing HEOS group. When the leader player is unjoined from a HEOS group the group will be destroyed. Example service data payload:
+
+```json
+{
+  "entity_id": "media_player.heos_player1",
```
Comment (frenck) [inline source/_integrations/heos.markdown]: Should be YAML.

File: source/_integrations/heos.markdown [hunk 4]
```diff
@@ -63,6 +64,38 @@ Use the sign-in service to sign the connected controller into a HEOS account so
...
+}
+```
+
+| Attribute              | Description
+| ---------------------- | ---------------------------------------------------------------- |
+| `entity_id`            | Removes all grouping from this media player. [Required]
```
Comment (frenck) [inline source/_integrations/heos.markdown]: Same as above, add a column for requirements of attributes.

**Review by frenck — CHANGES_REQUESTED** [review]
Thanks for the PR @klada!

I've left some comments, could you please take a look? Thanks! 👍

**Comment by klada** [comment]
@frenck Thanks for the input, it makes sense to bring the docs in line with the rest of the codebase. Right now I simply followed the current style in the HEOS docs. I think I have 3 options:

a) Use your suggested documentation style for my changes only
b) Update the entire HEOS documentation to the new documentation style, including my changes
c) Come up with a new PR (against the "current" branch?) first to address the style issues with the existing docs.

Which way do you want me to go?

**Comment by frenck** [comment]
I think fixing the current docs, rebasing this PR and update it to match would be the best option.

JSON input, for example, isn't used anymore and already afffects current users.

**Review by frenck — DISMISSED** [review]
Rebased the PR onto the latest `next` branch, converted JSON -> YAML and fixed the tables. 🎉 

✅ Approved. Can be merged as soon as the parent PR gets merged.

**Comment by frenck** [comment]
Replaced by #20428

**Pull request closed without merging**